### PR TITLE
Feature/new UI

### PR DESCRIPTION
**Pull request by dreitzner**
- [x] Iframe reload
- [x] one logic to rule them all
- [x] Change URLS back before merge
- [x] V2.0 for all

### CODE REVIEW

File: src/workfront/add-companyName-projectHeader.js [hunk 1]
```diff
@@ -14,40 +14,38 @@
...
+        if (window.wfGetOptions().showCompanyName) {                     
+            const elements = (await getElementsFromDocument('.thead.project-hours')) || [];
+            elements.forEach(getProjectFromWorkFront);
+        }
     }
 
-    function getProjectFromWorkFront(projectHTMLElement) {
-        return fetch(`https://emakina.my.workfront.com/attask/api/v12.0/proj/search?ID=${projectHTMLElement.getAttribute('data-projectid')}&fields=company:name`)
+    async function getProjectFromWorkFront(projectHTMLElement) {
+        return fetch(`https://emakina.sb01.workfront.com/attask/api/v12.0/proj/search?ID=${projectHTMLElement.getAttribute('data-projectid')}&fields=company:name`)
```
Comment (dreitzner) [inline src/workfront/add-companyName-projectHeader.js]: change to live URL

File: src/workfront/disable-completed-task.js [hunk 2]
```diff
@@ -36,7 +38,7 @@
     }
 
     async function fetchStatus(id) {
-        return fetch(`https://emakina.my.workfront.com/attask/api/v11.0/task/search?ID=${id}&fields=status`)
+        return fetch(`https://emakina.sb01.workfront.com/attask/api/v11.0/task/search?ID=${id}&fields=status`)
```
Comment (dreitzner) [inline src/workfront/disable-completed-task.js]: live URL

File: src/workfront/expand-project-header.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    // create a listener for all headers and save it to the local storage if it gets closed
+    async function getExpandedHeader()  {
+        const headers = await getElementsFromDocument('.thead.project-hours');
+        if(!headers) return;
+        
+        const allHeaders = [...headers];      
+        
+        allHeaders.forEach(header => {
+            header.addEventListener('click', function() {
```
Comment (dreitzner) [inline src/workfront/expand-project-header.js]: arrow function

File: src/workfront/expand-project-header.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    init();
+    
+    async function init(){
+        await setExpandedHeader();
+        getExpandedHeader();
+    }
+
+    // create a listener for all headers and save it to the local storage if it gets closed
+    async function getExpandedHeader()  {
+        const headers = await getElementsFromDocument('.thead.project-hours');
```
Comment (dreitzner) [inline src/workfront/expand-project-header.js]: move into getHeaderElements function

File: src/workfront/expand-project-header.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        if(!headers) return;
+        
+        const allHeaders = [...headers];      
+        
+        allHeaders.forEach(header => {
+            header.addEventListener('click', function() {
+                const headerId = header.getAttribute('data-projectid');
+                let closedHeaders = JSON.parse(localStorage.getItem(storageKey)) || [];
```
Comment (dreitzner) [inline src/workfront/expand-project-header.js]: class Store {
    constructor() {
        // read for first time
        this.object = localStorage.getItem('abc') || [];
    }

    get value () {
        return this.object;
    }

    set value (object) {
        this.object.push(object);
        localStorage.setItem('abc', this.object);
    }

    remove(obj) {
        this.object = this.object.filter(item => item.id !== obj.id)
    }
}

const store = new Store();

const worjwithThis = store.value
store.value = 'newVal';
store.remove({id: 'abc'})

File: src/workfront/include-scripts/check-if-new-or-old-UI.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+(function(window) {
+    'use strict';
+    let usesQuicksilver;
+
+    window.checkUI = () => { 
+        usesQuicksilver = window.config;
+    }
+
+    window.getUsesQuicksilver = () => {
+        return usesQuicksilver;
```
Comment (dreitzner) [inline src/workfront/include-scripts/check-if-new-or-old-UI.js]: return window.config

File: src/workfront/indicate-missing-comment.js [hunk 7]
```diff
@@ -14,76 +14,90 @@
...
+    const containerSelector = '#CommentPanel > menu';
+    const warningMessageSelector = '#CommentPanel > menu > p';
     const warningMessageStyle = 'color: tomato; padding: 15px 0; font-size: 1.2em; font-weight: bold;';
     const warningMessageText = 'Not all entries have a comment';
 
     const del = getSystemDecimalSeparator();
 
-    document.head.addEventListener('WF_RELOAD', init);
-    document.head.addEventListener('WF_NEW-TASK',e => initNewTask(e));
+    document.head.addEventListener('WF_NEW-TASK', e => initNewTask(e));
```
Comment (dreitzner) [inline src/workfront/indicate-missing-comment.js:32]: See if event is still triggered

File: src/workfront/indicate-missing-comment.js [hunk 8]
```diff
@@ -14,76 +14,90 @@
...
 
         // submit button is not always shown (on already commited ts)
         if(submitButton) {
             submitButton.disabled = emptyFieldFound;
         }
-        
-        emptyFieldFound ? warningMessage.classList.remove('hidden') : warningMessage.classList.add('hidden');
+
+        const oldWarningMessage = await getElementsFromDocument(`#${commentId}`);
+        emptyFieldFound ? (oldWarningMessage[0] || warningMessage)?.classList?.remove('hidden') : (oldWarningMessage[0] || warningMessage)?.classList?.add('hidden');
```
Comment (dreitzner) [inline src/workfront/indicate-missing-comment.js]: line breaks pls

File: src/workfront/indicate-missing-comment.js [hunk 9]
```diff
@@ -14,76 +14,90 @@
...
     }
 
     function checkAllCommentsAndMarkFields(elements) {
         let isEmptyCommentPresent = false;
-
+
```
Comment (dreitzner) [inline src/workfront/indicate-missing-comment.js:94]: remove empty space

File: src/workfront/indicate-missing-comment.js [hunk 10]
```diff
@@ -14,76 +14,90 @@
...
 
     function checkAllCommentsAndMarkFields(elements) {
         let isEmptyCommentPresent = false;
-
+        
         elements.forEach(e => {
             const comment = e.getAttribute('data-description');
             const value = e.value;
-
+
```
Comment (dreitzner) [inline src/workfront/indicate-missing-comment.js:98]: remove empty space

File: src/workfront/indicate-missing-comment.js [hunk 11]
```diff
@@ -107,12 +121,22 @@
...
+                {
+                    const val = e.value;
+                    checkAll(elements, warningMessage, submitButton);
+                    if (window.wfGetOptions().correctComma) { 
+                        const operation = shouldRoundToNearestQuarter() ? roundStringToNearestQtr : toSystemDecimalDelimiter;
+                        if (val) {
+                            e.value = operation(val);
+                        }
+                    }
+                    // TO-DO: change functionality to call when user has been idle for a while
```
Comment (dreitzner) [inline src/workfront/indicate-missing-comment.js]: todo?

File: src/workfront/indicate-missing-comment.js [hunk 12]
```diff
@@ -145,16 +169,8 @@
...
-        return document.getElements(selector);
-    }
-
-    function getElement(selector) {
-        return document.getElement(selector);
-    }
-
     function getSystemDecimalSeparator() {
-        var n = 1.1;
+        let n = 1.1;
```
Comment (dreitzner) [inline src/workfront/indicate-missing-comment.js]: const?

File: src/workfront/indicate-norm.js [hunk 13]
```diff
@@ -14,35 +14,40 @@
...
 
-        const col = createTableRows(data);
+        const col = await createTableRows(data);
+        if(!col) return;
 
         addListener(col, parseToFloat(data.extRefID));
     }
 
-    function createTableRows(data){
+    async function createTableRows(data){
```
Comment (dreitzner) [inline src/workfront/indicate-norm.js]: async?

File: src/workfront/indicate-norm.js [hunk 14]
```diff
@@ -103,13 +116,9 @@
     }
 
     function fetchProjectData(timesheetId) {
-        return fetch(`https://emakina.my.workfront.com/attask/api/v11.0/tshet/search?ID=${timesheetId}&fields=*`)
+        return fetch(`https://emakina.sb01.workfront.com/attask/api/v11.0/tshet/search?ID=${timesheetId}&fields=*`)
```
Comment (dreitzner) [inline src/workfront/indicate-norm.js]: live url

File: src/workfront/warning-current-timesheet.js [hunk 15]
```diff
@@ -93,19 +101,15 @@
         }
     }
 
-    function getCurrentTsId() {
-        return document.getElement('[data-timesheetid]').getAttribute('data-timesheetid');
-    }
-
     function getCurrentTs(currentTsId) {
-        return fetch(`https://emakina.my.workfront.com/attask/api/v11.0/tshet/search?ID=${currentTsId}&fields=status`)
+        return fetch(`https://emakina.sb01.workfront.com/attask/api/v11.0/tshet/search?ID=${currentTsId}&fields=status`)
```
Comment (dreitzner) [inline src/workfront/warning-current-timesheet.js]: url

File: src/workfront/warning-current-timesheet.js [hunk 16]
```diff
@@ -93,19 +101,15 @@
...
             .then(e => e.json())
             .then(e => e.data[0]);
     }
 
 
     function getOldestOpenTsBeforeToday() {
-        return fetch('https://emakina.my.workfront.com/attask/api/v11.0/tshet/search?endDate=$$TODAYb-1m&endDate_Mod=between&endDate_Range=$$TODAYe-1m&userID=$$USER.ID&userID_Mod=in&status=O&status_Mod=in&OR:1:endDate=$$TODAYbw&OR:1:endDate_Mod=lte&OR:1:userID=$$USER.ID&OR:1:userID_Mod=in&OR:1:status=O&OR:1:status_Mod=in&endDate_Sort=asc&$$LIMIT=1')
+        return fetch('https://emakina.sb01.workfront.com/attask/api/v11.0/tshet/search?endDate=$$TODAYb-1m&endDate_Mod=between&endDate_Range=$$TODAYe-1m&userID=$$USER.ID&userID_Mod=in&status=O&status_Mod=in&OR:1:endDate=$$TODAYbw&OR:1:endDate_Mod=lte&OR:1:userID=$$USER.ID&OR:1:userID_Mod=in&OR:1:status=O&OR:1:status_Mod=in&endDate_Sort=asc&$$LIMIT=1')
```
Comment (dreitzner) [inline src/workfront/warning-current-timesheet.js]: url

File: src/workfront/wf-combined.js [hunk 17]
```diff
@@ -1,28 +1,37 @@
...
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/include-scripts/check-if-new-or-old-UI.js
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/include-scripts/event-handler.js
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/include-scripts/access-document-object.js
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/include-scripts/convert-save-into-event.js
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/expand-project-header.js
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/highlight-currentday.js
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/warning-current-timesheet.js
+// @require      https://raw.githubusercontent.com/EmakinaBE/tampermonkey-scripts/feature/New-UI/src/workfront/add-companyName-projectHeader.js
```
Comment (dreitzner) [inline src/workfront/wf-combined.js]: no 🐫-case

File: src/workfront/expand-project-header.js (hunk 4 above)
Comment (Roupecs) [reply src/workfront/expand-project-header.js]: #done

File: src/workfront/expand-project-header.js (hunk 3 above)
Comment (Roupecs) [reply src/workfront/expand-project-header.js]: #done

File: src/workfront/expand-project-header.js (hunk 5 above)
Comment (Roupecs) [reply src/workfront/expand-project-header.js]: #done

File: src/workfront/indicate-missing-comment.js (hunk 8 above)
Comment (Toniiboni) [reply src/workfront/indicate-missing-comment.js]: #done

File: src/workfront/indicate-missing-comment.js (hunk 9 above)
Comment (Toniiboni) [reply src/workfront/indicate-missing-comment.js:94]: #done

File: src/workfront/indicate-missing-comment.js (hunk 10 above)
Comment (Toniiboni) [reply src/workfront/indicate-missing-comment.js:98]: #done

File: src/workfront/indicate-missing-comment.js (hunk 8 above)
Comment (Roupecs) [reply src/workfront/indicate-missing-comment.js:91]: #done

File: src/workfront/indicate-missing-comment.js (hunk 12 above)
Comment (Toniiboni) [reply src/workfront/indicate-missing-comment.js]: #done

File: src/workfront/indicate-norm.js (hunk 13 above)
Comment (Toniiboni) [reply src/workfront/indicate-norm.js]: #done

**Pull request merged**